### DISCUSSION TITLE
Coloring by PLDDT - using the dedicated Preset, and PluginContext (without React UI)

### DISCUSSION THREAD

**Discussion by 0gust1** [molstar/molstar#925] [Q&A]
Hi,

Context; I started a Svelte wrapper library for Molstar (**very early** alpha, not very stable / showable right now): https://github.com/0gust1/molstar-svelte. 

And I'm currently trying to reproduce the loading + coloring of an AlphaFold CIF file, and can't get it to work.

I took inspiration in the relevant parts of the Viewer app,

- QualityAssessmentPLDDTPreset and its usage: https://github.com/molstar/molstar/blob/fb16cd0070846fa4e42121961f888d5e73d3876c/src/apps/viewer/app.ts#L15C10-L15C38
- The dedicated loadAlphaFoldDb function and its behavior. https://github.com/molstar/molstar/blob/fb16cd0070846fa4e42121961f888d5e73d3876c/src/apps/viewer/app.ts#L303

But I may have missed something...

**Note:**
I need to keep a ref to the loaded Object to later remove it from the plugin, when my dedicated svelte componed unMounts/is destroyed, so I can't just copy paste the similar code from the Viewer.

**What I do:** 

Initialize Molstar "classically" ([very close from the docs](https://molstar.org/docs/plugin/#plugincontext-without-built-in-react-ui)).

Register the QualityAssessmentPLDDTPreset:
```ts
import { QualityAssessmentPLDDTPreset } from 'molstar/lib/extensions/model-archive/quality-assessment/behavior.js';
//...
plugin.builders.structure.representation.registerPreset(QualityAssessmentPLDDTPreset);
```

Load an alphafold cif file and try to apply the preset:  
```ts
data = await plugin.builders.data.download({ url: 'https://alphafold.ebi.ac.uk/files/AF-Q8W3K0-F1-model_v4.cif' }, { state: { isGhost: false } });
const trajectory = await plugin.builders.structure.parseTrajectory(data, 'mmcif');
const model = await plugin.builders.structure.createModel(trajectory);
const struct = await plugin.builders.structure.createStructure(model);
structure = struct;
await plugin.builders.structure.hierarchy.applyPreset(trajectory, 'preset-structure-representation-ma-quality-assessment-plddt');
```

I tried a lot of different ways to apply the preset (on the model, on the structure, etc.), I also tried to inject some params (coming from `DownloadStructure.createDefaultParams(plugin.state.data.root.obj!, plugin);`), but no luck.

I feel like I miss some further initialization/setup and/or my loading + preset applying is not right.

When I log the plugin instance, it seem that the preset was registred correctly:
<img width="975" alt="Capture d’écran 2023-09-21 à 22 10 10" src="https://github.com/molstar/molstar/assets/578154/f04950f6-7955-4c0a-a215-b492d8872862">

**Comment by dsehnal** [comment]
The pLDDT is a "representation" not "hierarchy" preset:

```ts
const plugin = this.plugin;
const data = await plugin.builders.data.download({ url: 'https://alphafold.ebi.ac.uk/files/AF-Q8W3K0-F1-model_v4.cif' }, { state: { isGhost: false } });
const trajectory = await plugin.builders.structure.parseTrajectory(data, 'mmcif');
const model = await plugin.builders.structure.createModel(trajectory);
const struct = await plugin.builders.structure.createStructure(model);
await plugin.builders.structure.representation.applyPreset(struct, QualityAssessmentPLDDTPreset.id as any);
```

**Reply by 0gust1** [reply]
Thanks @dsehnal ! 
You're right, damn, it was right before my eyes.

So right now, it's better, I see the structure, but the coloring is still the default. I'll dig further, but if you have any idea of what could be still missing, I'll take it!

<img width="485" alt="Capture d’écran 2023-09-22 à 17 44 07" src="https://github.com/molstar/molstar/assets/578154/9dc867e1-9246-42ee-8f85-cb2badeb9fe0">

The good news: now I have correctly setup this library project (in term of software factory), I can have a more structured approach learning and working with molstar (less "dirty" hack and slash). I also may have questions later on the differences between the various molstar internals/APIs.
In this library project, the documentation will be quite an important topic, and I plan to document as much I can when I will correctly figure out some topics (maybe the occasion to also contribute back to Molstar? I hope so).

**Reply by dsehnal** [reply]
I actually tested the code I posted in the "embedded.html" of the default viewer wrapper and it worked.

Maybe you can try copy-pasting the entire `Viewer` code and just using the `PluginContext` instead of the UI version? (And perhaps removing the extensions which contain references to react as a sanity test?
